Cargo.toml

```diff
@@ -130,7 +130,7 @@ serde = { version = "1.0.216", features = ["derive", "rc"] }
 serde_json = { version = "1.0.133", features = ["preserve_order"] }
 similar-asserts = { version = "1.6.0" }
 smallvec = { version = "1.7.0", features = ["union"] }
-stack-graphs = { version = "0.13.0" }
+stack-graphs = { path = "submodules/stack-graphs/stack-graphs", version = "0.14.0" }
 string-interner = { version = "0.17.0", features = [
```

Comment:
> the ability to rewind the arena allocator used for partial paths after resolving each reference

Do you think this would be useful for the upstream project? maybe suggesting it as a PR, in case they accept it? then we don't have to maintain the fork at all.

Reply:
We can try, but I doubt it's useful for their normal use cases. The main problem is it's not exactly safe, since you have no direct control over the mutability of the database and a mutable reference is required to do anything meaningful with it. That means you can accidentally allocate new objects in the partial paths arena (which are invalidated when you reset) that you'll be referencing in the database.

I think it may be possible to change the design to take an immutable database reference, but it's probably a much bigger change.